OnServiceChange/handler.ts

```diff
@@ -178,16 +179,42 @@ export const log = (d: unknown): void => {
   throw new Error(`To be implement ${d}`);
 };
 
-export const onInvalidDocument = (d: unknown): T.Task<void> => {
+export const onInvalidDocument = (
+  d: unknown,
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+): T.Task<void> => {
+  telemetryClient.trackEvent({
+    name: "selfcare.services.oninvaliddocument",
+    properties: {
+      documentId: (d as RetrievedService).serviceId,
+      message: "Invalid document received"
+    },
+    tagOverrides: { samplingEnabled: "false" }
+  });
   throw new Error(`To be implement ${JSON.stringify(d, null, 2)}`);
 };
 
-export const onIgnoredDocument = (_d: unknown): void => void 0;
+export const onIgnoredDocument = (
```

Comment:
consider move this tracking method in a dedicate module

Reply:
Ok, done with 16b04ec